File: frontend/src/types/snmp.ts
export interface SNMPMetrics {
  router_id: string
  hostname: string
  ip_address: string
  community: string
  reachable: boolean
  sys_descr: string | null
  sys_name: string | null
  uptime_seconds: number | null
  cpu_5min_percent: number | null
  mem_free_bytes: number | null
  if_number: number | null
  error: string | null
  polled_at: string
}

File: backend/app/schemas/snmp.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class SNMPPollRequest(BaseModel):
    router_id: uuid.UUID


class SNMPMetrics(BaseModel):
    router_id: uuid.UUID
    hostname: str
    ip_address: str
    community: str
    reachable: bool
    sys_descr: str | None
    sys_name: str | None
    uptime_seconds: int | None
    cpu_5min_percent: int | None
    mem_free_bytes: int | None
    if_number: int | None
    error: str | None
    polled_at: datetime
